src/api/java/com/minecolonies/api/colony/requestsystem/request/IRequest.java

```diff
@@ -0,0 +1,197 @@
+package com.minecolonies.api.colony.requestsystem.request;
+
+import com.google.common.collect.ImmutableCollection;
+import com.minecolonies.api.colony.requestsystem.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.RequestState;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+/**
+ * Used to represent requests, of type R, made to the internal market of the colony.
+ *
+ * @param <R> The type of request, eg ItemStack, FluidStack etc.
+ */
+public interface IRequest<R>
```

Comment:
This class is not supposed to be in this PR yet.
Just the Factories part and the IToken part (and there respective implementations)